### PR TITLE
Remove a few uses of fully qualified names that rely on a DMD bug

### PR DESCRIPTION
**Pull request by BorisCarvajal**
These fully-qualified names are not really exposed in the current scope but are accessible due to bug dlang/dmd#12215.

### CODE REVIEW

File: std/array.d [hunk 1]
```diff
@@ -3636,13 +3636,15 @@ if (isDynamicArray!A)
...
     }
 
+    import std.format : FormatSpec;
+
     /// ditto
     template toString(Writer)
     if (isOutputRange!(Writer, char))
     {
-        import std.format : FormatSpec;
+        //import std.format : FormatSpec; this doesn't work, see issue 13737
```
Comment (BorisCarvajal) [inline std/array.d:3645]: I need some advise here.
Should I comment out the line or just delete it?

**Review by RazvanN7 — COMMENTED** [review]
Except for the `abs` one, I don't understand why all the others are bugs. You are allowed to use the fully qualified name for an imported symbol. E.g:

```d
import a : name;
func(a.name);
```
is and should be perfectly legal.

File: std/algorithm/iteration.d [hunk 2]
```diff
@@ -786,9 +786,9 @@ private struct MapResult(alias fun, Range)
     string  s1 = "hello world!";
     dstring s2 = "日本語";
     dstring s3 = "hello world!"d;
-    auto ms1 = map!(std.ascii.toUpper)(s1);
-    auto ms2 = map!(std.ascii.toUpper)(s2);
-    auto ms3 = map!(std.ascii.toUpper)(s3);
+    auto ms1 = map!(toUpper)(s1);
+    auto ms2 = map!(toUpper)(s2);
+    auto ms3 = map!(toUpper)(s3);
```
Comment (RazvanN7) [inline std/algorithm/iteration.d:791]: `std.ascii.toUpper` is perfectly legal here.

File: std/algorithm/iteration.d [hunk 3]
```diff
@@ -5459,7 +5459,7 @@ private struct SplitterResult(alias isTerminator, Range)
         ["là", "dove", "terminava", "quella", "valle"]
     ));
     assert(equal(
-        splitter!(std.uni.isWhite)("là dove terminava quella valle"),
+        splitter!(isWhite)("là dove terminava quella valle"),
```
Comment (RazvanN7) [inline std/algorithm/iteration.d:5462]: ditto

**Comment by BorisCarvajal** [comment]
> Except for the `abs` one, I don't understand why all the others are bugs. You are allowed to use the fully qualified name for an imported symbol.

Not when it's a selective `import`.
https://dlang.org/spec/module.html#selective_imports

Check yourself: https://run.dlang.io/is/Lh649f

**Comment by RazvanN7** [comment]
Wow! I'm learning something new every day. This will definitely require a deprecation cycle.

File: std/array.d (hunk 1 above)
Comment (RazvanN7) [reply std/array.d:3645]: Just delete it.

**Review by RazvanN7 — APPROVED** [review]

**Comment by RazvanN7** [comment]
Delete the commented line and this is good to go.

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**